client/network/src/peer_info.rs

```diff
@@ -138,13 +138,16 @@ pub struct Node<'a>(&'a NodeInfo);
 
 impl<'a> Node<'a> {
 	/// Returns the endpoint of an established connection to the peer.
-	pub fn endpoint(&self) -> &'a ConnectedPoint {
-		&self.0.endpoints[0] // `endpoints` are non-empty by definition
+	///
+	/// Returns `None` if we are disconnected. It can happen when we are
+	/// disconnected, however the node is still in cache for `CACHE_EXPIRE` period.
```

Comment:
```suggestion
	/// Returns `None` if we are disconnected from the node.
```

I don't think it's worth going into details.